### PR TITLE
handle http proxy config for webhooks

### PR DESCRIPTION
**Pull request by bwagner5**
Issue #, if available:
https://github.com/aws/aws-node-termination-handler/issues/144

Description of changes:
Add support to proxy webhooks. This does not proxy kubernetes API requests or http calls to the EC2 Instance Metadata Service. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: config/helm/aws-node-termination-handler/README.md [hunk 1]
```diff
@@ -83,3 +84,5 @@ Parameter | Description | Default
 `securityContext.runAsUserID` | User ID to run the container | `1000`
 `securityContext.runAsGroupID` | Group ID to run the container | `1000` 
 
+
+
```
Comment (haugenj) [inline config/helm/aws-node-termination-handler/README.md:88]: ?

File: config/helm/ec2-metadata-test-proxy/templates/clusterrole.yaml [hunk 2]
```diff
@@ -9,29 +9,4 @@ rules:
...
-    - daemonsets
-  verbs:
-    - get
-- apiGroups:
-    - apps
-  resources:
-    - daemonsets
-  verbs:
-    - get
+
```
Comment (jaypipes) [inline config/helm/ec2-metadata-test-proxy/templates/clusterrole.yaml:12]: The above changes look unrelated?

Comment (bwagner5) [reply config/helm/ec2-metadata-test-proxy/templates/clusterrole.yaml:12]: It is unrelated, but when I was copying pasting the ec2-metadata-test-proxy helm chart, I noticed that neither of the test fixture charts needed the permissions. Note that both of these are only test fixtures and the only reason they are in helm is to make the cleanup easier in the e2e test suite.

Comment (jaypipes) [reply config/helm/ec2-metadata-test-proxy/templates/clusterrole.yaml:12]: yerah, makes sense :)
